src/gamepad/controller.cpp

```diff
@@ -24,6 +52,14 @@ void Controller::update() {
     this->LeftY = this->controller.get_analog(ANALOG_LEFT_Y);
     this->RightX = this->controller.get_analog(ANALOG_RIGHT_X);
     this->RightY = this->controller.get_analog(ANALOG_RIGHT_Y);
+
+    this->updateScreen();
+}
+
+void Controller::println(uint8_t line, std::string str, std::uint32_t duration) {
+    if (line > 2) std::exit(1); // TODO: change handling
+
+    screen_buffer[line].push_back(std::make_pair(str, duration));
```

Comment:
```suggestion
    screen_buffer[line].push_back({ .text = std::move(str), .duration = duration});
```
Use `std::move` for efficiency.